### PR TITLE
BFS Implementation in Java

### PR DESCRIPTION
**Pull request by ranesr**
This pull request is for issue number #25 

Plus
- Binary Search Implementation in Java
- gitignore file

### CODE REVIEW

File: Graph Theory/Breadth First Search Algorithm/Java/Graph.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	// Adjacency Lists
+	private LinkedList<Integer> adj[];
+
+	// Constructor for Graph
+	@SuppressWarnings("unchecked")
+	public Graph(int v) {
+		this.v = v;
+		adj = new LinkedList[v];
+		for (int i=0; i<v; ++i) {
```
Comment (VamsiSangam) [inline Graph Theory/Breadth First Search Algorithm/Java/Graph.java]: We really want spaces on either side of every binary operator in the whole file. It improves readability.

**Comment by VamsiSangam** [comment]
Please use only one pull request for each issue. Make a commit to remove the binary search implementation. You can add it via another PR. And please remove the ```.gitignore``` file too, we don't need it. :smile:

**Comment by ranesr** [comment]
`.gitignore` file really helps when multiple people are working on different systems. That's why I added it. But, if you really think it is not needed, I will remove it.

**Comment by VamsiSangam** [comment]
I will think about your advice, but for now please remove it.

**Comment by ranesr** [comment]
@VamsiSangam I have made the suggested changes. Please review the code.

**Review by VamsiSangam — APPROVED** [review]

**Comment by VamsiSangam** [comment]
Thanks for understanding! You should be able to see your code live on the [corresponding post](http://theoryofprogramming.com/2014/12/25/breadth-first-search-algorithm/) in a few minutes. :smile:

**Pull request merged**